frontend/src/containers/tables/ProgramCycle/NewProgramCycle/UpdateProgramCycle.tsx

```diff
@@ -0,0 +1,194 @@
+} from '@api/programCycleApi';
+import { useMutation } from '@tanstack/react-query';
+import type { DefaultError } from '@tanstack/query-core';
+import { useBaseUrl } from '@hooks/useBaseUrl';
+import { useSnackbar } from '@hooks/useSnackBar';
+
+interface UpdateProgramCycleProps {
+  program: ProgramQuery['program'];
+  programCycle?: ProgramCycle;
+  onClose: () => void;
+  onSubmit: () => void;
+  step?: string;
+}
+
+interface MutationError extends DefaultError {
+  data: any;
+}
+
+export const UpdateProgramCycle = ({
+  program,
+  programCycle,
+  onClose,
+  onSubmit,
+  step,
+}: UpdateProgramCycleProps) => {
+  const { t } = useTranslation();
+  const { businessArea } = useBaseUrl();
+  const { showMessage } = useSnackbar();
+
+  const validationSchema = Yup.object().shape({
+    end_date: Yup.date()
+      .required(t('End Date is required'))
+      .min(today, t('End Date cannot be in the past'))
+      .max(program.endDate, t('End Date cannot be after Programme End Date'))
+      .when(
+        'start_date',
+        (start_date, schema) =>
+          start_date &&
+          schema.min(
+            start_date,
+            `${t('End date have to be greater than')} ${moment(
+              start_date,
+            ).format('YYYY-MM-DD')}`,
+          ),
+      ),
+  });
+
+  const initialValues: {
+    [key: string]: string | boolean | number | null;
+  } = {
+    id: programCycle.id,
+    title: programCycle.title,
+    start_date: programCycle.start_date,
+    end_date: undefined,
+  };
+
+  const { mutateAsync, isPending, error } = useMutation<
+    ProgramCycleUpdateResponse,
+    MutationError,
+    ProgramCycleUpdate
+  >({
+    mutationFn: async (body) => {
+      return updateProgramCycle(
+        businessArea,
+        program.id,
+        decodeIdString(programCycle.id),
+        body,
+      );
+    },
+    onSuccess: () => {
+      onSubmit();
+    },
+  });
+
+  const handleSubmit = async (values: FormikValues) => {
+    try {
+      await mutateAsync({
+        title: programCycle.title,
+        start_date: programCycle.start_date,
+        end_date: values.end_date,
+      });
+      showMessage(t('Programme Cycle Updated'));
+    } catch (e) {
+      /* empty */
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({ submitForm, values }) => (
+        <Form>
+          <DialogTitleWrapper>
+            <DialogTitle>
+              <Box display="flex" justifyContent="space-between">
+                <Box>{t('Add New Programme Cycles')}</Box>
+                {step && <Box>{step}</Box>}
+              </Box>
+            </DialogTitle>
+          </DialogTitleWrapper>
+          <DialogContent>
+            <DialogDescription>
+              <GreyText>
+                {t(
+                  'Before you create a new Cycles, it is necessary to specify the end date of the existing Cycles',
```

Comment:
`                  'Before you create a new Cycle, it is necessary to specify the end date of the existing Cycle',
`